packages/flutter/lib/src/material/text_field.dart

```diff
@@ -775,6 +775,7 @@ class _TextFieldState extends State<TextField> with AutomaticKeepAliveClientMixi
     }
     _requestKeyboard();
     _confirmCurrentSplash();
+    _editableTextKey.currentState.hideToolbar();
```

Comment:
Fix 1/2: This fixes the problem where the selection menu could get stuck open.  I had to add a check to `hideToolbar` to make sure that we're not hiding a toolbar that's already hidden, but it seems like that probably should have been there anyway.

Reply:
It might be safer to turn it off at the start of the function? although I don't think we turn on toolbar in selectPosition

Reply:
I think I like putting it at the top more, I'll make that change.  It seems to work fine.